tests/frameworks/layers/l2l4_network_creation_test.py

```diff
@@ -281,37 +321,168 @@ def testMultipleL4L2ColumnLinks(self):
+      "externalInput_0.dataOut-->L4Column_0.basalGrowthCandidates",
+      "externalInput_1.dataOut-->L4Column_1.basalGrowthCandidates",
+      "externalInput_2.dataOut-->L4Column_2.basalGrowthCandidates"])
+
+    # This gets textual representations of the links.
+    links = set([link.second.getMoniker() for link in links])
+
+    # Build a descriptive error message to pass to the user
+    error_message = "Links incorrectly formed in multicolumn L2L4 network: \n"
+    for link in desired_links:
+      if not link in links:
+        error_message += "Failed to find link: {}\n".format(link)
+
+    for link in links:
+      if not link in desired_links:
+        error_message += "Found unexpected link: {}\n".format(link)
+
+    self.assertSetEqual(desired_links, links, error_message)
+
+
+  def testMultipleL4L2ColumnsWithTopologyCreate(self):
+    """
+    In this simplistic test we create a network with 5 L4L2Columns and
+    topological lateral connections, ensure it has the right number of regions,
+    and try to run some inputs through it without crashing.
+    """
+
+    net = createNetwork(networkConfig4)
+    self.assertEqual(len(net.regions.keys()), 20, "Incorrect number of regions")
+
+    # Add some input vectors to the queue
+    externalInput0 = net.regions["externalInput_0"].getSelf()
+    sensorInput0 = net.regions["sensorInput_0"].getSelf()
+    externalInput1 = net.regions["externalInput_1"].getSelf()
+    sensorInput1 = net.regions["sensorInput_1"].getSelf()
+    externalInput2 = net.regions["externalInput_2"].getSelf()
+    sensorInput2 = net.regions["sensorInput_2"].getSelf()
+    externalInput3 = net.regions["externalInput_3"].getSelf()
+    sensorInput3 = net.regions["sensorInput_3"].getSelf()
+    externalInput4 = net.regions["externalInput_4"].getSelf()
+    sensorInput4 = net.regions["sensorInput_4"].getSelf()
+
+    externalInput0.addDataToQueue([2, 42, 1023], 0, 9)
+    sensorInput0.addDataToQueue([2, 42, 1023], 0, 0)
+    externalInput1.addDataToQueue([2, 42, 1023], 0, 9)
+    sensorInput1.addDataToQueue([2, 42, 1023], 0, 0)
+    externalInput2.addDataToQueue([2, 42, 1023], 0, 9)
+    sensorInput2.addDataToQueue([2, 42, 1023], 0, 0)
+    externalInput3.addDataToQueue([2, 42, 1023], 0, 9)
+    sensorInput3.addDataToQueue([2, 42, 1023], 0, 0)
+    externalInput4.addDataToQueue([2, 42, 1023], 0, 9)
+    sensorInput4.addDataToQueue([2, 42, 1023], 0, 0)
+
+    # Run the network and check outputs are as expected
+
+    #import pdb; pdb.set_trace()
+    net.run(1)
+
+
+    # Spotcheck some of the phases
+    self.assertEqual(net.getPhases("externalInput_0"),(0,),
+                     "Incorrect phase externalInput_0")
+    self.assertEqual(net.getPhases("externalInput_1"),(0,),
+                     "Incorrect phase for externalInput_1")
+    self.assertEqual(net.getPhases("L4Column_0"),(2,),
+                     "Incorrect phase for L4Column_0")
+    self.assertEqual(net.getPhases("L4Column_1"),(2,),
```

Comment:
Done.  This was also an issue in some of the preexisting tests, and I've fixed it there as well.